### DISCUSSION TITLE
请问解码出来的文字，可以输出每个文字的置信度么？

### DISCUSSION THREAD

**Discussion by zcswdt** [PaddlePaddle/PaddleSpeech#586] [General]
您好，请问识别出来的文字，可以输出文字的置信度吗？

**Comment by zh794390558** [comment]
这个得修改解码的代码，还需要考虑word的置信度策略，是平均呢还是加和。比如 asssb, dsss_c中 s是取最大，还是累加，或者平均。具体如何做不是很清楚，可以看看有没有相关的文章。

**Reply by zcswdt** [reply]
> 这个得修改解码的代码，还需要考虑word的置信度策略，是平均呢还是加和。比如 asssb, dsss_c中 s是取最大，还是累加，或者平均。具体如何做不是很清楚，可以看看有没有相关的文章。

都可以，如果知道了单个字符的置信度，就可以知道整句话的置信度，所以单个字符的置信度不知道在哪里可以知道。

**Reply by zh794390558** [reply]
整句是有分数得，单词的分数没有。看下decoder得wrapper。

**Reply by zcswdt** [reply]
> 整句是有分数得，单词的分数没有。看下decoder得wrapper。

谢谢回复，请问整句的分数需要改哪个地方的代码才能显示出来？我想最终的目的是输入一句语音，然后识别成文字，给出整句话文字的概率的置信度。

**Reply by zhangyifei1** [reply]
> 整句是有分数得，单词的分数没有。看下decoder得wrapper。

整句话的分数在哪看？
